dace/codegen/targets/cpp.py

```diff
@@ -1374,11 +1487,18 @@ def array_interface_variable(var_name: str,
             # Throw a KeyError if this pointer also doesn't exist
             dispatcher.defined_vars.get(ptr_out, ancestor)
             # Otherwise use it
-            return ptr_out
+            result = ptr_out
         else:
-            return ptr_in
+            result = ptr_in
     else:
         # We might call this before the variable is even defined (e.g., because
         # we are about to define it), so if the dispatcher is not passed, just
         # return the appropriate string
-        return ptr_out if is_write else ptr_in
+        result = ptr_out if is_write else ptr_in
+    #Append the interface id, if provided
+    if interface_id is not None:
+        if isinstance(interface_id, tuple):
+            result = f"{result}_{interface_id[accessed_subset]}"
```

Comment:
I think it's nicer to leave it here, because usually at the callsite you don't really know wheter the interface_id you are passing is an integer or a list, so you'd need to do this check there. Would it work to make those two parameters a tuple instead?